src/random_ray/flat_source_domain.cpp

```diff
@@ -247,12 +251,38 @@ double FlatSourceDomain::compute_k_eff(double k_eff_old) const
       sr_fission_source_new += nu_sigma_f * scalar_flux_new_[idx];
     }
 
-    fission_rate_old += sr_fission_source_old * volume;
-    fission_rate_new += sr_fission_source_new * volume;
+    // Compute total fission rates in FSR
+    sr_fission_source_old *= volume;
+    sr_fission_source_new *= volume;
+
+    // Accumulate totals
+    fission_rate_old += sr_fission_source_old;
+    fission_rate_new += sr_fission_source_new;
+
+    // Store total fission rate in the FSR for Shannon calculation
+    p[sr] = sr_fission_source_new;
   }
 
   double k_eff_new = k_eff_old * (fission_rate_new / fission_rate_old);
 
+  double H = 0.0;
+  // defining an inverse sum for better performance
+  double inverse_sum = 1 / fission_rate_new;
+
+#pragma omp parallel for reduction(+ : H)
+  for (int i = 0; i < n_source_regions_; i++) {
```

Comment:
I'm noticing that we have two for loops in this function doing the same loop over all source regions, but are using two different iterator variable names (`sr` vs. `i`). I think we tend to use `sr` pretty consistently for looping over source regions through the code, so might as well use that for the second loop as well.

Reply:
This is a good convention! I have made this edit.